Taggers/plugins/DoubleHTagProducer.cc

```diff
@@ -848,6 +900,143 @@ namespace flashgg {
         float NNscore = outputs[0].matrix<float>()(0, 0);
         return NNscore;
     }
+
+    enum TheRunEra{y2016B,y2016C,y2016D,y2016E,y2016F,y2016G,y2016H,y2017B,y2017C,y2017D,y2017E,y2017F,y2018A,y2018B,y2018C,y2018D,y2016MC,y2017MC,y2018MC};
+
+    std::pair<double,double> DoubleHTagProducer::METXYCorr_Met_MetPhi(double uncormet, double uncormet_phi, int runnb, unsigned int year, bool isMC, int npv){
```

Comment:
Are you sure there is no tool in CMSSW already provided by JetMET?

Reply:
No, the integration in CMSSW is still going on. The deatils are here https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETRun2Corrections#xy_Shift_Correction_MET_phi_modu

The macro is here, which is being used by JME for now and we have used exact function to calculate corrections
https://lathomas.web.cern.ch/lathomas/METStuff/XYCorrections/XYMETCorrection.h

Reply:
The MET corrections should be implemented in a separate class, under Tagger/interface (src/). The DoubleHTags (ggHH and VBFHH) producers should call an instanc of the class. The year meta variable should have a more detailed name, like "XYMETcorr_MCyear".

Reply:
I have implemented this for ggHH Tag. Should I add the Mjj correction part in VBFHH as well? and if yes then after adding if I want to test the setup for VBFHH, I should set only one flag true of VBFHH?

Reply:
I have added L2 regression for VBFHH Tagger as well